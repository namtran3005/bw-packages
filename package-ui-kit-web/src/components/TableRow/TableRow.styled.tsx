import * as React from 'react';
import styled from 'styled-components';
import { TableRow as MuiTableRow } from '@material-ui/core';
import { TableRowProps as MuiTableRowProps } from '@material-ui/core/TableRow';
import { Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';

import { Root as TableCell } from '../TableCell';

interface RootProps extends MuiTableRowProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  background?: ColorsType;
  backgroundAlpha?: number;
  foreground?: ColorsType;
  highlighted?: boolean;
  highlight?: ColorsType;
  dimmed?: boolean;
}

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable @typescript-eslint/no-unused-vars */
  highlight,
  highlighted,
  background,
  backgroundAlpha,
  foreground,
  dimmed,
  /* eslint-enable @typescript-eslint/no-unused-vars */
  ...cleanProps
}) => <MuiTableRow {...cleanProps} />;

const defaultBackground = 'white';
const defaultAlpha = 0;

const getTableRowBackground = (props: RootProps) => {
  if (props.background) {
    if (props.backgroundAlpha !== undefined) {
      return Colors.getWithAlpha(props.background, props.backgroundAlpha);
    } else {
      return Colors.get(props.background);
    }
  }
  return Colors.getWithAlpha(defaultBackground, defaultAlpha);
};

const Root = styled(RootComponent)`
  &&& {
    background: ${getTableRowBackground};
    height: auto;
    color: ${({ foreground }) => Colors.get(foreground || 'primaryGrey')};

    ${TableCell} {
      border-bottom-width: 2px;
      border-bottom-color: ${({ highlighted, highlight }) =>
        highlighted
          ? Colors.get(highlight || 'primaryLilac')
          : Colors.get('primaryGrey')};
    }
  }
`;

Root.displayName = 'TableRow';
RootComponent.displayName = 'TableRowCompoment';

export { Root, RootProps, RootComponent };
