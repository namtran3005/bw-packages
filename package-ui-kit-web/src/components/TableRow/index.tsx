export { TableRow, TableRowProps } from './TableRow';
export { Root, RootProps, RootComponent } from './TableRow.styled';
