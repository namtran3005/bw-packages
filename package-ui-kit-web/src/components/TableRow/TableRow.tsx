import * as React from 'react';

import { Root, RootProps as TableRowProps } from './TableRow.styled';

const TableRow: React.FunctionComponent<TableRowProps> = (props) => {
  return <Root {...props} />;
};

export { TableRow, TableRowProps };
