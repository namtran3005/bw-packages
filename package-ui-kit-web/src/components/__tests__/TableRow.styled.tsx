import * as React from 'react';
import { shallow } from 'enzyme';
import { Root, RootComponent } from '../TableRow/TableRow.styled';

describe('TableRow', () => {
  it('should render table row', () => {
    const wrapper = shallow(<Root />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render table row root component', () => {
    const wrapper = shallow(<RootComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render table row with background', () => {
    const wrapper = shallow(<Root background="primaryLilac" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render table row with background and alpha', () => {
    const wrapper = shallow(
      <Root background="primaryLilac" backgroundAlpha={0.5} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
