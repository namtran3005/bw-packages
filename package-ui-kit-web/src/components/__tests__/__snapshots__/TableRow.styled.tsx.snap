// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should render table row 1`] = `
.c0.c0.c0 {
  background: rgba(255,255,255,0);
  height: auto;
  color: #D4D2D2;
}

.c0.c0.c0 .sc-bdfBwQ {
  border-bottom-width: 2px;
  border-bottom-color: #D4D2D2;
}

<TableRowCompoment
  className="c0"
/>
`;

exports[`TableRow should render table row root component 1`] = `<WithStyles(ForwardRef(TableRow)) />`;

exports[`TableRow should render table row with background 1`] = `
.c0.c0.c0 {
  background: #BEAAFF;
  height: auto;
  color: #D4D2D2;
}

.c0.c0.c0 .sc-bdfBwQ {
  border-bottom-width: 2px;
  border-bottom-color: #D4D2D2;
}

<TableRowCompoment
  background="primaryLilac"
  className="c0"
/>
`;

exports[`TableRow should render table row with background and alpha 1`] = `
.c0.c0.c0 {
  background: rgba(190,170,255,0.5);
  height: auto;
  color: #D4D2D2;
}

.c0.c0.c0 .sc-bdfBwQ {
  border-bottom-width: 2px;
  border-bottom-color: #D4D2D2;
}

<TableRowCompoment
  background="primaryLilac"
  backgroundAlpha={0.5}
  className="c0"
/>
`;
